$OpenBSD: patch-lib_Support_Unix_Threading_inc,v 1.3 2019/02/11 05:24:16 jca Exp $

Add OpenBSD support to be able to get the thread name

Index: lib/Support/Unix/Threading.inc
--- lib/Support/Unix/Threading.inc.orig
+++ lib/Support/Unix/Threading.inc
@@ -203,6 +203,12 @@ void llvm::get_thread_name(SmallVectorImpl<char> &Name
   ::pthread_getname_np(::pthread_self(), buf, len);
 
   Name.append(buf, buf + strlen(buf));
+#elif defined(__OpenBSD__)
+  constexpr uint32_t len = get_max_thread_name_length_impl();
+  char buf[len];
+  ::pthread_get_name_np(::pthread_self(), buf, len);
+
+  Name.append(buf, buf + strlen(buf));
 #elif defined(__linux__)
 #if HAVE_PTHREAD_GETNAME_NP
   constexpr uint32_t len = get_max_thread_name_length_impl();
